### DISCUSSION TITLE
Allow to configure `--env-mode=loose` in `turbo.json`

### DISCUSSION THREAD

**Discussion by NatoBoram** [vercel/turborepo#8611] [Ideas]
### Goals

Not clear why this is not already an option, as the desired effect is to have consistent behavior across all envs.

### Non-goals

Make it harder to transition from Turbo 1 to Turbo 2 in large monorepos by not having this option

### Background

* https://github.com/vercel/turbo/issues/6481

### Proposal

Adding `envMode` setting to `turbo.json`

**Comment by Hacksore** [comment]
I just upgraded from turbo `v1` to `v2` and found that it broke my workflow with 1password. Seems very unfortunate that I have to pass `--env-mode loose` explicitly to turbo commands instead of a global config setting in `turbo.json`.

I very much support this proposed ask ❤️.

https://github.com/Hacksore/1password-cli-env

**Reply by Snailedlt** [reply]
You can just add this to your `turbo.json`, to give every task access to every env variable. Which afaik is the same thing that `--env-mode=loose` does:

```json
    "globalEnv": [
        "*"
    ],
```

**Comment by Hacksore** [comment]
I just realized an option you have available today is adding env vars to `globalEnv`.

https://github.com/Hacksore/1password-cli-env/commit/bb1258300d8d333a902b0da21255e75cec7cfe22

cc @anthonyshew

**Comment by anthonyshew** [comment]
This is being worked on in https://github.com/vercel/turbo/pull/8757. Sorry for not seeing this earlier!

**Reply by Snailedlt** [reply]
Yup, so now that it's merged you can do this:
```json
    "globalEnv": [
        "*"
    ],
```

**Reply by anthonyshew** [reply]
We don't recommend this. This would make you miss cache for every task whenever any environment variable changes.

There's a small chance this is your intent, but not likely.

**Comment by anthonyshew** [comment]
This has shipped! https://github.com/vercel/turborepo/pull/8757
